src/libxively/io/fs/posix/xi_fs_posix.c

```diff
@@ -15,7 +18,19 @@ xi_state_t xi_fs_stat( const void* context,
     XI_UNUSED( context );
     XI_UNUSED( resource_type );
 
-    return xi_bsp_io_fs_stat( resource_name, resource_stat );
+    if ( NULL == resource_stat || NULL == resource_name )
+    {
+        return XI_INVALID_PARAMETER;
+    }
+    
+    xi_bsp_io_fs_stat_t bsp_io_fs_resource_stat;
+    xi_state_t result = xi_fs_bsp_io_fs_2_xi_state( xi_bsp_io_fs_stat( resource_name, &bsp_io_fs_resource_stat ) );
```

Comment:
yeah, so the implementation of this converter function should be in this file. At least I think so.

Reply:
Yep! Thanks!  👍